From 6e83c39adc1cc606279ae777d0e750e335d23047 Mon Sep 17 00:00:00 2001
From: Erik Schumacher <erik.schumacher@iris-sensing.com>
Date: Tue, 14 Mar 2023 09:56:44 +0100
Subject: [PATCH] lm75.c: Defer probe if i2c communication fails with EAGAIN

The lm75 sits on the same i2c bus as the epc660 which is blocking the
bus until the epc660 probe function configures the device. If a i2c
call fails during lm75 probe, we simply defer the probing.

Signed-off-by: Erik Schumacher <erik.schumacher@iris-sensing.com>
Signed-off-by: Jasper Orschulko <jasper@fancydomain.eu>
---
 drivers/hwmon/lm75.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/hwmon/lm75.c b/drivers/hwmon/lm75.c
index 5b2ea05c951e..e10ac02f889e 100644
--- a/drivers/hwmon/lm75.c
+++ b/drivers/hwmon/lm75.c
@@ -626,6 +626,8 @@ static int lm75_probe(struct i2c_client *client)
 	status = i2c_smbus_read_byte_data(client, LM75_REG_CONF);
 	if (status < 0) {
 		dev_dbg(dev, "Can't read config? %d\n", status);
+		if (status == -EAGAIN)
+			return -EPROBE_DEFER;
 		return status;
 	}
 	data->orig_conf = status;
-- 
2.46.0

